script.js

```diff
@@ -381,48 +387,46 @@ function generateCourseTable(parentElement, thisTest, classUrl) {
             document.querySelector(`#dataRow${parentElement}${i}`).appendChild(tableElements[num])
           }
           else {
-            var waldoApiUrl = `https://api.metalab.csun.edu/waldo/1.0/rooms?room=${location}`
-            console.log(waldoApiUrl)
-
-            var latitude
-            var longitude
-            var googleMapsURL = `https://www.google.com/maps/search/?api=1&query=${latitude},${longitude}`
-
-            const fetchWaldo = fetch(waldoApiUrl)
-              .then(response => response.json())
-              .then(data => {
-                latitude = data.rooms[0].latitude
-                longitude = data.rooms[0].longitude
-                googleMapsURL = `https://www.google.com/maps/search/?api=1&query=${latitude},${longitude}`
-                return googleMapsURL
-              });
-
-            const printAddress = (tableElements, num, location, parentElement, i) => {
-              fetchWaldo.then((a) => {
-                console.log(a);
-                if (num == 7) {
-                  tableElements[num].innerHTML = `<a href='${a}' target="_blank">${location}</a>`
-                  document.querySelector(`#dataRow${parentElement}${i}`).appendChild(tableElements[num])
-                }
-                else {
-                  tableElements[num].innerHTML = tableDataContent[num]
-                  document.querySelector(`#dataRow${parentElement}${i}`).appendChild(tableElements[num])
-                }
-              });
-            };
-            // tableElements[num].innerHTML = tableDataContent[num]
-            // document.querySelector(`#dataRow${parentElement}${i}`).appendChild(tableElements[num])
-            printAddress(tableElements, num, location, parentElement, i);
+            if (num == 7) {
+              tableElements[num].id = `location_${parentElement}${i}`
+              tableElements[num].innerHTML = tableDataContent[num]
+              document.querySelector(`#dataRow${parentElement}${i}`).appendChild(tableElements[num])
+
+              var waldoApiUrl = `https://api.metalab.csun.edu/waldo/1.0/rooms?room=${location}`
+              console.log(waldoApiUrl)
+
+              var latitude
```

Comment:
`let` & `const` can replace `var` declarations

Reply:
@BrianLinggadjaja What benefit does using `let` & `const` instead of `var` provide?